karma.conf.js

```diff
@@ -0,0 +1,266 @@
+				{
+					type: 'lcovonly',
+					dir: coverageDir
+				}
+			]
+		};
+
+		webpackConfig.module.rules.push( {
+			test: /\.jsx?$/,
+			loader: 'istanbul-instrumenter-loader',
+			include: /src/,
+			exclude: [
+				/node_modules/
+			],
+			query: {
+				esModules: true
+			}
+		} );
+	}
+
+	if ( options.sourceMap ) {
+		karmaConfig.preprocessors[ 'tests/**/*.jsx' ].push( 'sourcemap' );
+
+		webpackConfig.devtool = 'inline-source-map';
+	}
+
+	config.set( karmaConfig );
+};
+
+/**
+ * Returns the value of Karma's browser option.
+ *
+ * @param {Array.<String>} browsers
+ * @returns {Array.<String>|null}
+ */
+function getBrowsers( browsers ) {
+	if ( !browsers ) {
+		return null;
+	}
+
+	const newBrowsers = browsers
+		.map( browser => {
+			if ( browser !== 'Chrome' ) {
+				return browser;
+			}
+
+			return process.env.TRAVIS ? 'CHROME_TRAVIS_CI' : 'CHROME_LOCAL';
+		} );
+
+	if ( shouldEnableBrowserStack() ) {
+		return newBrowsers;
+	}
+
+	// If the BrowserStack is disabled, all browsers that start with a prefix "BrowserStack" should be filtered out.
+	// See: https://github.com/ckeditor/ckeditor5-dev/issues/358 and https://github.com/ckeditor/ckeditor5-dev/issues/402.
+	return newBrowsers.filter( browser => !browser.startsWith( 'BrowserStack' ) );
+}
+
+function shouldEnableBrowserStack() {
+	if ( !process.env.BROWSER_STACK_USERNAME ) {
+		return false;
+	}
+
+	if ( !process.env.BROWSER_STACK_ACCESS_KEY ) {
+		return false;
+	}
+
+	// If the repository slugs are different, the pull request comes from the community (forked repository).
+	// For such builds, BrowserStack will be disabled. Read more: https://github.com/ckeditor/ckeditor5-dev/issues/358.
+	return ( process.env.TRAVIS_EVENT_TYPE !== 'pull_request' || process.env.TRAVIS_PULL_REQUEST_SLUG === process.env.TRAVIS_REPO_SLUG );
+}
+
+/**
+ * @param {Array.<String>} args CLI arguments and options.
+ * @returns {Object} options
+ * @returns {Array.<String>} options.browsers Browsers that will be used to run tests.
+ * @returns {String} options.reporter A reporter that will presents tests results.
+ * @returns {Boolean} options.watch Whether to watch the files.
+ * @returns {Boolean} options.coverage Whether to generate code coverage.
+ * @returns {Boolean} options.sourceMap Whether to add source maps.
+ */
+function parseArguments( args ) {
+	const minimist = require( 'minimist' );
+
+	const config = {
+		string: [
+			'browsers',
+			'reporter'
+		],
+
+		boolean: [
+			'watch',
+			'coverage',
+			'source-map'
+		],
+
+		alias: {
+			b: 'browsers',
+			c: 'coverage',
+			r: 'reporter',
+			s: 'source-map',
+			w: 'watch',
+		},
+
+		default: {
+			browsers: 'Chrome',
+			reporter: 'mocha',
+			watch: false,
+			coverage: false,
+			'source-map': false,
+		}
+	};
+
+	const options = minimist( args, config );
+
+	options.sourceMap = options[ 'source-map' ];
+	options.browsers = options.browsers.split( ',' );
+
+	// Delete all aliases because we don't want to use them in the code.
```

Comment:
?

Reply:
?

Reply:
Ok. Didn't know that aliases are the part of minimist's results.